File: capy/src/e2e-cfun/java/dev/capylang/test/MinLiteralValueTest.java
package dev.capylang.test;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class MinLiteralValueTest {
    @Test
    void minIntCompilesAndExecutes() {
        assertThat(MinLiteralValue.minInt()).isEqualTo(Integer.MIN_VALUE);
    }

    @Test
    void minIntWorksInRegularExpressions() {
        assertThat(MinLiteralValue.minIntDigits()).isEqualTo(10);
    }

    @Test
    void minLongCompilesAndExecutes() {
        assertThat(MinLiteralValue.minLong()).isEqualTo(Long.MIN_VALUE);
    }
}

File: capy/src/e2e-cfun/js/dev/capylang/test/MinLiteralValue.test.js
const test = require('node:test');
const { assert, generatedModule } = require('./_support');

test('MinLiteralValue', () => {
    const min = generatedModule('dev/capylang/test/MinLiteralValue.js');
    assert.equal(min.minInt(), -2147483648);
    assert.equal(min.minIntDigits(), 10);
    assert.equal(min.minLong(), -9223372036854775808n);
    assert.equal(min.maxLong(), 9223372036854775807n);
    assert.equal(min.maxSafeIntegerLong(), 9007199254740991n);
    assert.equal(min.longAddition(), 9223372036854775807n);
    assert.equal(min.longOverflow(), -9223372036854775808n);
});
